contracts/src/warpDiscordBot/namesService/write/registerUser.ts

```diff
@@ -0,0 +1,28 @@
+import { validateInputArgumentPresence, validateString } from '../../../utils';
+import { ContractAction, ContractState, ContractResult, usersPrefix } from '../../types/types';
+
+declare const ContractError;
+
+export const registerUser = async (
+  state: ContractState,
+  { input: { id, address } }: ContractAction
+): Promise<ContractResult> => {
+  validateInputArgumentPresence(id, 'id');
+  validateString(id, 'id');
+  validateInputArgumentPresence(address, 'address');
+  validateString(address, 'address');
+
+  const users = state.users;
+
+  if (Object.keys(users).includes(id)) {
+    throw new ContractError('Id already assigned.');
+  }
+
+  if ([...Object.values(users)].includes(address)) {
```

Comment:
I believe this `users` object might be huge, so creating an array on the fly may cost some memory/time...did you test the performance in case of large object (e.g. 15k entries)?